RecoLocalTracker/SiPhase2VectorHitBuilder/src/VectorHitBuilderEDProducer.cc

```diff
@@ -0,0 +1,130 @@
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderEDProducer.h"
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderAlgorithm.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+VectorHitBuilderEDProducer::VectorHitBuilderEDProducer(edm::ParameterSet const& conf)
+    : offlinestubsTag_(conf.getParameter<std::string>("offlinestubs")),
+      maxOfflinestubs_(conf.getParameter<int>("maxVectorHits")),
+      algoTag_(conf.getParameter<std::string>("Algorithm")),
+      //_clusterProducer(conf.getParameter<edm::InputTag>("Clusters")),
+      readytobuild_(false) {
+  clusterProducer_ =
+      consumes<edmNew::DetSetVector<Phase2TrackerCluster1D>>(edm::InputTag(conf.getParameter<std::string>("Clusters")));
+
+  produces<edmNew::DetSetVector<Phase2TrackerCluster1D>>("ClustersAccepted");
+  produces<edmNew::DetSetVector<Phase2TrackerCluster1D>>("ClustersRejected");
+  produces<VectorHitCollectionNew>(offlinestubsTag_ + "Accepted");
+  produces<VectorHitCollectionNew>(offlinestubsTag_ + "Rejected");
+  setupAlgorithm(conf);
+}
+
+VectorHitBuilderEDProducer::~VectorHitBuilderEDProducer() { delete stubsBuilder_; }
+
+void VectorHitBuilderEDProducer::produce(edm::Event& event, const edm::EventSetup& es) {
+  LogDebug("VectorHitBuilderEDProducer") << "VectorHitBuilderEDProducer::produce() begin";
+
+  // get input clusters data
+  edm::Handle<edmNew::DetSetVector<Phase2TrackerCluster1D>> clustersHandle;
+  event.getByToken(clusterProducer_, clustersHandle);
+
+  // create the final output collection
+  std::unique_ptr<edmNew::DetSetVector<Phase2TrackerCluster1D>> outputClustersAccepted(
```

Comment:
prefer to use `make_unique` to avoid `new `

Reply:
Done